src/internal/snapshot/recovery.go

```diff
@@ -214,6 +215,65 @@ func (s *Snapshotter) dumpTx(ctx context.Context, tx *pachsql.Tx, opts CreateSna
 	return id, fh, nil
 }
 
+func (s *Snapshotter) pinDatabaseDump(ctx context.Context, fh io.Reader, id SnapshotID) (retErr error) {
+	ctx, done := log.SpanContext(ctx, "pinDatabaseDump")
+	defer done(log.Errorp(&retErr))
+
+	var closedFileSet, pinnedFileSet bool
+	fw := s.Storage.NewWriter(ctx)
```

Comment:
Fwiw, it is generally better to use the unordered writer because it handles unordered writes and bounds the memory usage. Using a writer is fine if only one file is being written and it can fit in memory, but it could be problematic if more files are added later.

Reply:
Done.  I think this probably always fits in memory but we are dumping to /tmp to avoid that constraint, so using the unordered writer makes sense here.